tracing-subscriber/src/fmt/writer.rs

```diff
@@ -651,12 +651,6 @@ pub struct Tee<A, B> {
 #[derive(Debug)]
 pub struct MutexGuardWriter<'a, W>(MutexGuard<'a, W>);
 
-/// Implements [`std::io::Write`] for an [`Arc`]<W> where `&W: Write`.
```

Comment:
This warned about never being constructed. As far as I can tell it is not actually used for the current impl of `MakeWriter` for `Arc<W>`.

Reply:
Hmm I don't get this error. The `ArcWriter` is also public, so it's probably used by someone. I think it's better if don't change this.

Reply:
It's `pub` within this module, but it's not exported from this module:

```
djc-2021 v0.1.x tracing-subscriber $ rg ArcWriter
src/fmt/writer.rs
658:pub struct ArcWriter<W>(Arc<W>);
1135:// === impl ArcWriter ===
1137:impl<W> io::Write for ArcWriter<W>
```

Reply:
Ah, whoops; if the `MakeWriter` implementation for `Arc` doesn't actually use this, I think getting rid of it is correct.

Reply:
> It's `pub` within this module, but it's not exported from this module

Yes it is?
Maybe I'm stupid, but:
https://github.com/tokio-rs/tracing/blob/6d00d7d9f72dc6797138a1062bc33073afbad5a1/tracing-subscriber/src/fmt/writer.rs#L658
and
https://github.com/tokio-rs/tracing/blob/6d00d7d9f72dc6797138a1062bc33073afbad5a1/tracing-subscriber/src/fmt/mod.rs#L201
make it available. Now it probably isn't used anywhere as there's a comment explaining it's an implementation detail.

Reply:
Hmm, you're right. So I guess we might want to add `#[doc(hidden)]` on it, or something?

Reply:
Sure, and maybe a `#[deprecated(...)]` as well?

Reply:
Done.